src/main/java/uk/gov/homeoffice/digital/sas/balancecalculator/kafka/consumer/BalanceCalculatorConsumerService.java

```diff
@@ -0,0 +1,101 @@
+import static uk.gov.homeoffice.digital.sas.balancecalculator.constants.Constants.ACTUATOR_KAFKA_FAILURE_DESCRIPTION;
+import static uk.gov.homeoffice.digital.sas.balancecalculator.constants.Constants.ACTUATOR_KAFKA_FAILURE_URL;
+import static uk.gov.homeoffice.digital.sas.balancecalculator.constants.Constants.KAFKA_SUCCESSFUL_DESERIALIZATION;
+import static uk.gov.homeoffice.digital.sas.balancecalculator.constants.Constants.KAFKA_UNSUCCESSFUL_DESERIALIZATION;
+import static uk.gov.homeoffice.digital.sas.kafka.constants.Constants.SCHEMA_JSON_ATTRIBUTE;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.gson.reflect.TypeToken;
+import io.micrometer.core.instrument.Counter;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.kafka.annotation.KafkaListener;
+import org.springframework.messaging.handler.annotation.Payload;
+import org.springframework.stereotype.Service;
+import org.springframework.util.ObjectUtils;
+import uk.gov.homeoffice.digital.sas.balancecalculator.actuator.ActuatorCounters;
+import uk.gov.homeoffice.digital.sas.balancecalculator.models.TimeEntry;
+import uk.gov.homeoffice.digital.sas.kafka.consumer.KafkaConsumerService;
+import uk.gov.homeoffice.digital.sas.kafka.message.KafkaEventMessage;
+
+@Service
+@Slf4j
+@Getter
+@ComponentScan({
+    "uk.gov.homeoffice.digital.sas.kafka.consumer",
+    "uk.gov.homeoffice.digital.sas.kafka.validators",
+    "uk.gov.homeoffice.digital.sas.balancecalculator.actuator"})
+public class BalanceCalculatorConsumerService {
+
+  private ObjectMapper mapper = new ObjectMapper();
+
+
+  private Counter errorCounter;
+  private TimeEntry timeEntry;
+  private final String kafkaSupportedResourceName;
+
+  private final KafkaConsumerService<TimeEntry> kafkaConsumerService;
+
+  private final ActuatorCounters counters;
+
+  private KafkaEventMessage<TimeEntry> kafkaEventMessage;
+
+  public BalanceCalculatorConsumerService(@Value("${kafka.supported.resource.name}")
+                                          String kafkaSupportedResourceName,
+                                          KafkaConsumerService<TimeEntry> kafkaConsumerService,
+                                          ActuatorCounters counters) {
+    this.kafkaSupportedResourceName = kafkaSupportedResourceName;
+    this.kafkaConsumerService = kafkaConsumerService;
+    this.counters = counters;
+  }
+
+  @KafkaListener(
+      topics = {"${spring.kafka.template.default-topic}"},
+      groupId = "${spring.kafka.consumer.group-id}"
+  )
+  public void onMessage(@Payload String message) {
+    errorCounter = counters.setUpCounters(ACTUATOR_KAFKA_FAILURE_URL, ACTUATOR_ERROR_TYPE,
```

Comment:
Does this mean we set up the counter on every message?

Reply:
Moved that initialization into the constructor.